CODE/usb_history.py

```
@@ -0,0 +1,89 @@
+    def _enum_subkeys(root, path, warn_func):
+        """Yield all subkeys of a registry key, logging warnings on errors."""
+        try:
+            with winreg.OpenKey(root, path) as key:
+                for i in range(1024):  # reasonable upper bound to avoid infinite loops
+                    try:
+                        yield winreg.EnumKey(key, i)
+                    except OSError as e:
+                        if e.winerror == 259:  # no more data
+                            break
+                        warn_func(f"Error enumerating {path} index {i}: {e}")
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)